components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -4527,6 +4527,59 @@ public int compare(Application o1, Application o2) {
         return applications;
     }
 
+    /**
+     * Returns all the consumerkeys of application which are subscribed for the given api
+     *
+     * @param identifier APIIdentifier
+     * @return Consumerkeys
+     * @throws org.wso2.carbon.apimgt.api.APIManagementException if failed to get Applications for given subscriber.
+     */
+    public String[] getConsumerKeys(APIIdentifier identifier) throws APIManagementException {
+
+        Set<String> consumerKeys = new HashSet<String>();
+
+        Connection connection = null;
+        PreparedStatement prepStmt = null;
+        ResultSet rs = null;
+
+        int apiId = -1;
+        String sqlQuery = "SELECT " +
+                          "   MAP.CONSUMER_KEY " +
+                          "FROM " +
+                          "   AM_SUBSCRIPTION SUB, " +
+                          "   AM_APPLICATION_KEY_MAPPING MAP " +
+                          "WHERE " +
+                          "   SUB.APPLICATION_ID = MAP.APPLICATION_ID " +
+                          "   AND SUB.API_ID = ?";
+
+        try {
+
+            connection = APIMgtDBUtil.getConnection();
+            apiId = getAPIID(identifier, connection);
+
+            prepStmt = connection.prepareStatement(sqlQuery);
+            prepStmt.setInt(1, apiId);
+            rs = prepStmt.executeQuery();
+
+            while (rs.next()) {
+                consumerKeys.add(rs.getString("CONSUMER_KEY"));
+            }
+
+        } catch (SQLException e) {
+            handleException("Error when reading application subscription information", e);
+        } finally {
+            APIMgtDBUtil.closeAllConnections(prepStmt, connection, rs);
+        }
+
+        String[] consumerKeyArray;
+        if (consumerKeys.size() == 0) {
```

Comment:
Should we not be able to get rid of this redundant "if" condition? Returning an empty array is anyway recommended rather than returning null.